src/game_api/texture.cpp

```diff
@@ -274,34 +276,40 @@ void reload_texture(const char** texture_name)
 
     auto renderer_ptr = render.renderer();
     auto load_texture = *vtable_find<LoadTextureFunT*>(renderer_ptr, c_LoadTextureVirtualIndex);
+
+    if (texture_name->dx_resource)
+    {
+        texture_name->dx_resource->release(); // should be safe since apparently the objects have reference counter, essentially working like shared pointer
+        game_free(texture_name->dx_resource);
+    }
     load_texture(renderer_ptr, texture_name);
```

Comment:
This needs more testing, hard to tell if this is correct since game never reloads textures

But the `load_texture` call does create new `dx_resource` object and just overrides the old one without even checking (normally `dx_resource` is set to the placeholder texture when `Resource` is created, and then this function makes proper object for the texture and overrides the placeholder)